main.tf

```diff
@@ -0,0 +1,197 @@
+provider "aws" {
+  region = "${var.region}"
+}
+
+locals {
+  default_tags = {
+    Project     = "${var.project}"
+    Environment = "${var.environment}"
+  }
+
+  acm_certificate_arn = "${var.acm_certificate_arn == "" ? module.aws-cert.acm_certificate_arn : var.acm_certificate_arn}"
+  distribution_label  = "${var.project}-${var.environment}-distribution_label"
+}
+
+module "vpc" {
+  source = "github.com/lean-delivery/tf-module-awscore"
+
+  create_route53_zone = "true"
+  root_domain         = "${var.root_domain}"
+  project             = "${var.project}-${var.environment}"
+  vpc_cidr            = "${var.vpc_cidr}"
+
+  tags = "${merge(local.default_tags, var.tags)}"
+}
+
+module "aws-cert" {
+  source = "github.com/lean-delivery/tf-module-aws-acm"
+
+  module_enabled = "${var.acm_certificate_arn == "" ? true : false}"
+  domain         = "${var.root_domain}"
+  zone_id        = "${module.vpc.route53_zone_id}"
+
+  tags = "${merge(local.default_tags, var.tags)}"
+}
+
+resource "aws_cloudfront_origin_access_identity" "origin_access_identity" {
+  comment = "${local.distribution_label}"
+}
+
+resource "aws_s3_bucket" "origin" {
+  bucket        = "${var.s3_bucket_name}"
+  acl           = "private"
+  force_destroy = true
+
+  policy = <<EOF
+{
+"Id": "bucket_policy_site",
+"Version": "2012-10-17",
+"Statement": [
+    {
+    "Sid": "bucket_policy_site_root",
+    "Action": ["s3:ListBucket"],
+    "Effect": "Allow",
+    "Resource": "arn:aws:s3:::${var.s3_bucket_name}",
+    "Principal": {"AWS":"${aws_cloudfront_origin_access_identity.origin_access_identity.iam_arn}"}
+    },
+    {
+    "Sid": "bucket_policy_site_all",
+    "Action": ["s3:GetObject"],
+    "Effect": "Allow",
+    "Resource": "arn:aws:s3:::${var.s3_bucket_name}/*",
+    "Principal": {"AWS":"${aws_cloudfront_origin_access_identity.origin_access_identity.iam_arn}"}
+    }
+]
```

Comment:
better move this policy into a template

Reply:
ok, [done](https://github.com/lean-delivery/tf-module-static-artifact-storage/pull/1/commits/373c84d41c978a933bb230a9249ab4e5d6582dea)